adapter/service/outlvlswitch/routing.go

```diff
@@ -0,0 +1,161 @@
+				return nil, fmt.Errorf("adapter: error while setting level: %w", err)
+			}
+
+			_, err = outLvlSwitch.SendLevelReport(true)
+			if err != nil {
+				return nil, fmt.Errorf("adapter: error while sending level report: %w", err)
+			}
+
+			_, err = outLvlSwitch.SendBinaryReport(true)
+			if err != nil {
+				return nil, fmt.Errorf("adapter: error while sending binary report: %w", err)
+			}
+
+			return nil, nil
+		}),
+	)
+}
+
+// RouteCmdBinarySet returns a routing responsible for handling the command.
+func RouteCmdBinarySet(adapter adapter.Adapter) *router.Routing {
+	return router.NewRouting(
+		HandleCmdBinarySet(adapter),
+		router.ForService(OutLvlSwitch),
+		router.ForType(CmdBinarySet),
+	)
+}
+
+// HandleCmdBinarySet returns a handler responsible for handling the command.
+func HandleCmdBinarySet(adapter adapter.Adapter) router.MessageHandler {
+	return router.NewMessageHandler(
+		router.MessageProcessorFn(func(message *fimpgo.Message) (*fimpgo.FimpMessage, error) {
+			s := adapter.ServiceByTopic(message.Topic)
+			if s == nil {
+				return nil, fmt.Errorf("adapter: service not found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			outLvlSwitch, ok := s.(Service)
+			if !ok {
+				return nil, fmt.Errorf("adapter: incorrect service found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			binary, err := message.Payload.GetBoolValue()
+			if err != nil {
+				return nil, err
+			}
+
+			err = outLvlSwitch.SetBinary(binary)
+			if err != nil {
+				return nil, fmt.Errorf("adapter: error while setting binary: %w", err)
+			}
+
+			_, err = outLvlSwitch.SendBinaryReport(true)
+			if err != nil {
+				return nil, fmt.Errorf("adapter: error while sending binary report: %w", err)
+			}
+
+			return nil, nil
+		}),
+	)
+}
+
+// RouteCmdLvlGetReport returns a routing responsible for handling the command.
+func RouteCmdLvlGetReport(adapter adapter.Adapter) *router.Routing {
+	return router.NewRouting(
+		HandleCmdLvlGetReport(adapter),
+		router.ForService(OutLvlSwitch),
+		router.ForType(CmdLvlGetReport),
+	)
+}
+
+// HandleCmdLvlGetReport returns a handler responsible for handling the command.
+func HandleCmdLvlGetReport(adapter adapter.Adapter) router.MessageHandler {
+	return router.NewMessageHandler(
+		router.MessageProcessorFn(func(message *fimpgo.Message) (*fimpgo.FimpMessage, error) {
+			s := adapter.ServiceByTopic(message.Topic)
+			if s == nil {
+				return nil, fmt.Errorf("adapter: service not found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			outLvlSwitch, ok := s.(Service)
+			if !ok {
+				return nil, fmt.Errorf("adapter: incorrect service found under the provided address: %s", message.Addr.ServiceAddress)
+			}
+
+			_, err := outLvlSwitch.SendLevelReport(true)
+			if err != nil {
+				return nil, fmt.Errorf("adapter: error while sending level report: %w", err)
+			}
+
+			_, err = outLvlSwitch.SendBinaryReport(true)
+			if err != nil {
```

Comment:
I am not sure if you need to send binary report with level report too... Let's consult with @chribru tomorrow.